### ISSUE TITLE
Crash CocoaMQTTReader.frameReady()

### ISSUE THREAD

**Issue by adicatur** [emqx/CocoaMQTT#305]
Hi @HJianBo 

I get this issue in `CocoaMQTTREader.freameReady()` could you please check this, the number of this crash quite high :)

![Image of frame ready](https://d1edrlpyc25xu0.cloudfront.net/kiwari-prod/image/upload/fszcEnNZIo/Screen+Shot+2019-12-05+at+16.57.39.png)

Regards

**Comment by HJianBo** [comment]
Hi, @adicatur Which version do you use? And could you describe the brief steps or scenario to help to reproduce it? Thanks

**Comment by adicatur** [comment]
Hi @HJianBo 

Thanks for the response, we use `1.2.5`. Unfortunately, it's hard to reproduce, we got this from report third party. Do you find anything from this image?

**Comment by HJianBo** [comment]
Nor I can't get any more ideas from this image.  To be honest, the 1.2.x has more and more `force unwarpping` operation. It hides a lot of risk of collapse.

I will refactor all of this in v1.3.0 as soon as possible. Hope it will work in the future🤣

**Comment by adicatur** [comment]
Haha 🙏  thanks for noticing this, btw @HJianBo would you like to share your plan when do you release this :))

**Comment by HJianBo** [comment]
emmmm...I'll try to release it by the beginning of next month. 

The 1.3.0-alpha.1 is more stable for v1.2.x I think. Maybe you can try it. Please let me know if you have any questions

**Comment by adicatur** [comment]
Hi @HJianBo, thanks for the response, will try, but I guess someone has tried and still find an issue :) https://github.com/emqx/CocoaMQTT/issues/302

**Comment by adicatur** [comment]
Hi, @HJianBo just want to make sure, so when we use 1.3.x we will not see this issue since you are refactoring this part? CMIIW

**Comment by HJianBo** [comment]
Yes, I have refactored a lot of modules to delete the `unwarp` operations.

This bug may be fixed. But I have not more and more tests. And a recent effort on the broker projects.

So It should work well. But I can't completely sure

**Comment by adicatur** [comment]
Hi @HJianBo  thanks for the response, we see the minimum is using iOS 10, what's the behind reason not to use iOS 9?
